### PR TITLE
sdtm.oak SDTM spec, Algorithms Documentation and template r code for cm domain

### LINKED ISSUE #17: Documentation:  Template R code
### Please select a category the issue is focused on?

Developer Guides

### Let us know where something needs a refresh or put your idea here!

Develop a Vignette to demonstrate how the {sdtm.oak} functions and algorithms can be used to create SDTM domains.

### PR DESCRIPTION
**Pull request by rammprasad**
Thank you for your Pull Request! We have developed this task checklist from the
[Development Process
Guide](https://pharmaverse.github.io/admiraldev/articles/development_process.html)
to help with the final steps of the process. Completing the below tasks helps to
ensure our reviewers can maximize their time on your code as well as making sure
the oak codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task
or check off that it is not relevant to your Pull Request. This checklist is
part of the Github Action workflows and the Pull Request will not be merged into
the `devel` branch until you have checked off each task.

- [x] Place Closes #17  into the beginning of your Pull z
Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the 
[tidyverse style guide](https://style.tidyverse.org/). Run 
`styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should
consider realistic data scenarios and edge cases, e.g. empty datasets, errors,
boundary cases etc. - See 
[Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you 
fully follow the 
[deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords
and families. Refer to the 
[categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the
`NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it
has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build oak site `pkgdown::build_site()` and check that all affected
examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: inst/create_cm_template.R [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+#
+
+library(sdtm.oak)
+library(dplyr)
+
+# Read Specification
+
+# sdtm_spec <- read_sdtm_spec(filename = "~/study/study_sdtm_spec.csv")
+
+study_ct <- read_study_ct(filename = "~/study/study_sdtm_ct.csv")
```
Comment (yli110-stat697) [inline inst/create_cm_template.R:19]: should we have a template ct file along with the package? Just to get a feeling of the expected input

File: inst/create_cm_template.R [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+# Read in raw data
+
+MD1 <- read_raw_data_csv(filename = "~/study/MD1.csv") |>
```
Comment (yli110-stat697) [inline inst/create_cm_template.R:23]: Is `MD1.csv` the long lean format we agreed on? Or we changed the input format at some point?

File: inst/create_cm_template.R [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  ) |>
+  dplyr::mutate(
+    STUDYID = "test_study",
+    DOMAIN = "CM",
+    CMCAT = "GENERAL CONMED"
+  ) |>
+  derive_usubjid() |>
+  derive_sequence(keys = c(USUBJID, CMTRT)) |>
+  derive_visit_visitnum() |>
+  derive_study_day(
```
Comment (yli110-stat697) [inline inst/create_cm_template.R:223]: I see how this is related to my function `calculate_study_day` I was working on. So we will have another function `derive_study_day` as a wrapper or should I should modify my function to suite this purpose here?

File: inst/create_cm_template.R (hunk 2 above)
Comment (rammprasad) [reply inst/create_cm_template.R:23]: We can skip the long, lean format for now. Based on my research, we can develop the algorithms as they are.

File: inst/create_cm_template.R (hunk 3 above)
Comment (rammprasad) [reply inst/create_cm_template.R:223]: Updated the parameters to match `calculate_study_day`

**Comment by rammprasad** [comment]
@ramiromagno - One of the iso8601 test cases fail in the rcmd check PR. Will you be able to check the pipeline and take a look?

**Comment by ramiromagno** [comment]
> @ramiromagno - One of the iso8601 test cases fail in the rcmd check PR. Will you be able to check the pipeline and take a look?

Hello Ram, yes, I will look into it!

File: inst/create_cm_template.R (hunk 1 above)
Comment (kamilsi) [reply inst/create_cm_template.R]: I fully support your suggestion of including a template file with the package. In the realm of clinical trials, particularly non-commercial ones, we often encounter a vast array of data formats. A template, serving as a standard reference for expected inputs, could greatly assist users in structuring their data appropriately for the {sdtm.oak} package.

I understand that this is a future consideration and not necessary for the current PR.

File: vignettes/articles/study_sdtm_spec.Rmd [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+properties of data collection fields in the study.
+
+- Data Dictionaries: Identifier and the controlled terms collected at the
+source.
+
+- Visits: Name of the visits as defined in the EDC.
+
+**eDT Metadata**
+
+eDT Metadata is the blueprint metadata that describes the data collected as part of that external data transfer. This includes
```
Comment (kamilsi) [inline vignettes/articles/study_sdtm_spec.Rmd:73]: To improve clarity, you could consider adding a brief clarification to specify where the data is transferred to or provide more context about the purpose of the external data transfer.

File: vignettes/articles/study_sdtm_spec.Rmd [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+
+- Dataset name, label, repeating properties, etc.
```
Comment (kamilsi) [inline vignettes/articles/study_sdtm_spec.Rmd:75]: What are "repeating properties"?

File: vignettes/articles/study_sdtm_spec.Rmd [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+
+- Variable name, datatype, label and associated codelist, etc.
```
Comment (kamilsi) [inline vignettes/articles/study_sdtm_spec.Rmd:77]: Does eDT metadata describe the eCRF metadata? Can it be completely derived from it? Or does it have some unique information?

**Review by kamilsi — COMMENTED** [review]
I want to express my sincere appreciation for your efforts in making `sdtm.oak` more approachable. This is a crucial step in ensuring that documentation is accessible to a broader audience.

Having worked as a developer for many years on SDTM, I can personally attest to the challenges posed by the numerous acronyms and specialized terminology within this domain. Even with my experience, I still find myself struggling with some of the many, many acronyms present here.

Given this, I wholeheartedly support the @ramiromagno idea of creating a dictionary, as he suggested on Slack. Such a resource would undoubtedly be a valuable asset, not only for newcomers but also for seasoned professionals like myself. It would greatly enhance our understanding of the content and make `sdtm.oak` even more user-friendly.

File: inst/cm_domain/create_cm_template.R [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+  derive_oak_id_vars()
+
+# Create CM domain. The first step in creating CM domain is to create the topic variable
+
+cm <- cm_daw_data |>
+  # Derive topic variable
+  assign_no_ct(
+    raw_dataset = MD1, # This is added for pseudocode. Not required as pipe will send it
+    raw_variable = MDRAW,
+    target_sdtm_var = CMTRT
```
[the remark this answers is not in the document]
Comment (edgar-manukyan) [reply inst/cm_domain/create_cm_template.R:34]: Thanks Kamil, perhaps this can explained in the centralize article/dictionary that @ramiromagno was suggesting?

File: vignettes/articles/study_sdtm_spec.Rmd [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+that needs to be collected during a study.
+
+## Standards Metadata
+
+The standards metadata used in {sdtm.oak} is sourced from the CDISC Library or 
+sponsor MDR or any other form of documentation where standards are maintained. 
+This metadata provides information on the following:
+
+- The relationship between Data Collection Standards (eCRF & eDT), SDTM mapping, 
+and Controlled Terminology
```
[the remark this answers is not in the document]
Comment (edgar-manukyan) [reply vignettes/articles/study_sdtm_spec.Rmd:45]: Perhaps centralized article/dictionary can cover this and all other abbreviations we use in the package?

**Review by kamilsi — APPROVED** [review]

**Review by edgar-manukyan — APPROVED** [review]

File: vignettes/articles/study_sdtm_spec.Rmd (hunk 5 above)
Comment (edgar-manukyan) [reply vignettes/articles/study_sdtm_spec.Rmd:75]: We will address this in #38

File: vignettes/articles/study_sdtm_spec.Rmd (hunk 4 above)
Comment (edgar-manukyan) [reply vignettes/articles/study_sdtm_spec.Rmd]: I added a bit information about it.
